code-server: ros2 initial release

greinet opened the pull request:
Adds ROS2 to code-server and openvscode-server.

greinet commented:
I added the s6v2 files but i don know how to test them.  
As I don't know too much about s6-overlayI assume the file permissions should be okay, otherwise I can also change them:
```
$ tree -pufi root/etc
[drwxr-xr-x abc     ]  root/etc
[drwxr-xr-x abc     ]  root/etc/cont-init.d
[-rwxr-xr-x abc     ]  root/etc/cont-init.d/95-apt-get
[-rwxr-xr-x abc     ]  root/etc/cont-init.d/98-ros2
[drwxr-xr-x abc     ]  root/etc/s6-overlay
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/init-mod-code-server-ros2-add-package
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/init-mod-code-server-ros2-add-package/dependencies.d
[-rw-r--r-- abc     ]  root/etc/s6-overlay/s6-rc.d/init-mod-code-server-ros2-add-package/dependencies.d/init-mods
[-rwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/init-mod-code-server-ros2-add-package/run
[-rw-r--r-- abc     ]  root/etc/s6-overlay/s6-rc.d/init-mod-code-server-ros2-add-package/type
[-rw-r--r-- abc     ]  root/etc/s6-overlay/s6-rc.d/init-mod-code-server-ros2-add-package/up
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/init-mods-package-install
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/init-mods-package-install/dependencies.d
[-rw-r--r-- abc     ]  root/etc/s6-overlay/s6-rc.d/init-mods-package-install/dependencies.d/init-mod-code-server-ros2-add-package
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/user
[drwxr-xr-x abc     ]  root/etc/s6-overlay/s6-rc.d/user/contents.d
[-rw-r--r-- abc     ]  root/etc/s6-overlay/s6-rc.d/user/contents.d/init-mod-code-server-ros2-add-package
```

aptalca commented on root/etc/cont-init.d/98-ros2:
```diff
@@ -0,0 +1,19 @@
+    curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | sudo tee /etc/apt/sources.list.d/ros2.list > /dev/null
+
+    echo "ros-humble-ros-base ros-dev-tools" >> /mod-repo-packages-to-install.list
```
v2 doesn't use the package install script later on so this line should do the `apt install` instead

aptalca commented on root/etc/cont-init.d/98-ros2:
```diff
@@ -0,0 +1,19 @@
+    apt-get update && apt-get install -y gnupg
+fi
+
+if [ ! -f "/etc/apt/sources.list.d/ros2.list" ]; then
```
repo addition should be moved to a separate init before `95-apt-get` like `94-ros2-src` similar to here: https://github.com/linuxserver/docker-mods/blob/code-server-nodejs/root/etc/cont-init.d/94-nodejs-src

aptalca reviewed (changes requested):
The perms look good, just see the comments about the repo addition and apt install in v2 script

greinet commented:
Thanks for the detailed instructions, I moved the repo addition and changed to use `apt install`. The one thing I wasn't able to solve was where to put 
```printf "\n# Automatically source ROS2 humble\nsource /opt/ros/humble/setup.bash" >> /etc/bash.bashrc```  
It automatically sources ros2 so the command is available in the terminal, i guess it's not strictly needed but would be helpful to include.

aptalca commented:
@greinet the `abc` user's home folder is set to `/config` so the profile and bashrc should be in there

aptalca commented on root/etc/cont-init.d/94-ros2-src:
```diff
@@ -0,0 +1,12 @@
+fi
+
+source /etc/lsb-release
+[[ ! -f "/etc/apt/sources.list.d/ros2.list" ]] && \
```
Please change this to an if block. Otherwise the container will fail on a subsequent start because this part will exit 1 and since it's the last line in this init script, the init script will exit 1, failing the whole init

greinet replied:
I changed this to an if block and moved the ros2 sourcing into the `.bashrc` inside the home folder.

aptalca commented:
Apologies, I totally missed that you pushed the requested changes. I'll make some modifications to it based on our latest changes and then merge. Thanks for your patience.
